platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1026 @@
+   - If it exists, send the candidate roster to the `TssMessageCreator` and `TssMessageValidator`.
+   - Send all `TssMessageTransactions` to the `TssMessageValidator`.
+   - Send all `TssVoteTransactions` to the `TssMessageValidator`.
+   - Wait for the pubic shares and ledger id to come back for the active roster.
+   - Verify that the returned ledger id matches what exists in the state.
+     - NOTE: handling changes to the LedgerId are addressed in the `TSS-Ledger-ID-Updates` proposal.
+     - If there was no Ledger Id in the state, set the ledger Id in the state.
+2. Handle `TssMessageTransaction` system transactions.
+   - Handle items in consensus order.
+   - Determine the next sequence number to use for the `TssMessageMap` key.
+   - Insert into the `TssMessageMap` if it is correct to do so.
+     - Don’t insert multiple messages for the same share index.
+   - If successfully inserted into the `TssMessageMap`, send the `TssMessageTransaction`
+     to the `TssMessageValidator`.
+3. Handle `TssVoteTransaction` system transactions.
+   - If a threshold number of votes (totaling at least 1/3 of weight), all with the same vote byte
+     array, have already been received for the candidate roster, discard the `TssVoteTransaction`.
+   - Insert into the `TssVoteMap` if it is correct to do so.
+     - Don’t insert multiple votes from the same node; discard it if it is redundant.
+   - Send the `TssVoteTransaction` to the TSS Message Validator.
+4. At end of round handling.
+   - Detect if there is a new candidate roster set in the state.
+   - Send the candidate roster to the `TssMessageCreator` and `TssMessageValidator`.
+   - If the previous candidate roster has been replaced, use its saved roster hash to clear the
+     roster map of the old candidate roster and clear the `TssMessageMap` and `TssVoteMap` of
+     entries related to the roster hash.
+
+#### TSS Message Creator
+
+The TSS message creator is responsible for generating a node’s TSSMessages for its shares. This runs as its own
+asynchronous thread since this may be a computationally expensive operation.
+
+Internal Elements:
+
+1. active roster hash
+2. candidate roster hash
+3. Map<RosterHash, Roster> rosters
+4. Map<RosterHash, List<PairingPrivateKey>> privateShares
+5. NodeId (constructor)
+6. TssEncryptionKey (constructor)
+
+Inputs:
+
+1. Active Roster (Wire)
+2. Candidate Roster (Wire)
+3. private shares (Wire)
+
+`On Active Roster`:
+
+1. If the new active roster is the same as the existing one, do nothing.
+2. If the new active roster == the candidate roster,
+   1. clear the candidate roster hash.
+   2. clear the maps of the previous active roster data.
+3. If there existed a previous active roster and the candidate roster is different from the new active roster, log an
+   error.
+4. Set the new active roster
+5. If the active roster has private shares set, and there exists a candidate roster,
```

Comment:
Why would TssMessageTransactions be created an there is a new active roster? I think this should only be done when there is a new candidate roster.